### PR TITLE
Connect the gas-button-group component to redux and a live api.

### PR DESCRIPTION
**Pull request by danjm**
This PR connects the GasPriceButtonGroup to live data from the ethgasstation api.

Note two architectural choices:
(1) The data from state is already formatted and renderable when passed to the component. No operations are made on the data in the `*.component.js` files
(2) Only the `GasModalPageContainer` is connected to state. Data is directly passed to its children and its children's children.

buttongroupconnected

### CODE REVIEW

**Review by whymarrh — DISMISSED** [review]
Two defaults that might be worth adjusting

**Comment by whymarrh** [comment]
On the two choices:

> The data from state is already formatted and renderable when passed to the component. No operations are made on the data in the `*.component.js` files

I'm cool with this, it seems fine in this PR for sure

> Only the `GasModalPageContainer` is connected to state. Data is directly passed to its children and its children's children

Again, in the context of this PR, it looks pretty good

File: ui/app/components/gas-customization/gas-modal-page-container/basic-tab-content/basic-tab-content.component.js [hunk 1]
```diff
@@ -15,7 +15,13 @@ export default class BasicTabContent extends Component {
     return (
       <div className="basic-tab-content">
         <div className="basic-tab-content__title">Suggest gas fee increases</div>
-        <GasPriceButtonGroup {...this.props.gasPriceButtonGroupProps} />
+        <GasPriceButtonGroup
+          {...this.props.gasPriceButtonGroupProps}
```
[the remark this answers is not in the document]
Comment (danjm) [reply ui/app/components/gas-customization/gas-modal-page-container/basic-tab-content/basic-tab-content.component.js]: addressed in https://github.com/MetaMask/metamask-extension/pull/5214/commits/5132fff633837026b8cf7c2decdc7fa8d7a85330

File: ui/app/components/button-group/button-group.component.js [hunk 2]
```diff
@@ -16,7 +17,9 @@ export default class ButtonGroup extends PureComponent {
   }
 
   state = {
-    activeButtonIndex: this.props.defaultActiveButtonIndex || 0,
+    activeButtonIndex: this.props.noButtonActiveByDefault
+      ? null
+      : this.props.defaultActiveButtonIndex || 0,
```
[the remark this answers is not in the document]
Comment (danjm) [reply ui/app/components/button-group/button-group.component.js]: addressed in https://github.com/MetaMask/metamask-extension/pull/5214/commits/5132fff633837026b8cf7c2decdc7fa8d7a85330

File: ui/app/components/gas-customization/gas-modal-page-container/basic-tab-content/basic-tab-content.component.js [hunk 3]
```diff
@@ -15,7 +15,13 @@ export default class BasicTabContent extends Component {
     return (
       <div className="basic-tab-content">
         <div className="basic-tab-content__title">Suggest gas fee increases</div>
-        <GasPriceButtonGroup {...this.props.gasPriceButtonGroupProps} />
+        <GasPriceButtonGroup
+          className="gas-price-button-group"
+          noButtonActiveByDefault={true}
+          showCheck={true}
+          handleGasPriceSelection={(newPrice) => console.log('New Price:', newPrice)}
```
Comment (whymarrh) [inline ui/app/components/gas-customization/gas-modal-page-container/basic-tab-content/basic-tab-content.component.js]: Nit: `console.log`

File: ui/app/components/gas-customization/gas-modal-page-container/gas-modal-page-container.container.js [hunk 4]
```diff
@@ -51,6 +29,7 @@ const mapDispatchToProps = dispatch => {
     hideModal: () => dispatch(hideModal()),
     updateCustomGasPrice: (newPrice) => dispatch(setCustomGasPrice(newPrice)),
     updateCustomGasLimit: (newLimit) => dispatch(setCustomGasLimit(newLimit)),
+    handleGasPriceSelection: newPrice => console.log('NewPrice: ', newPrice),
```
Comment (whymarrh) [inline ui/app/components/gas-customization/gas-modal-page-container/gas-modal-page-container.container.js]: Nit: `console.log`

**Review by whymarrh — CHANGES_REQUESTED** [review]
Two quick nitpicks about `console.log`s that we should probably remove

**Comment by danjm** [comment]
@whymarrh I forgot that console.logs are removed in a downstream PR. Merging

**Pull request merged**